pilot/pkg/serviceregistry/external/servicediscovery.go

```diff
@@ -112,8 +112,8 @@ func NewServiceDiscovery(configController model.ConfigStoreCache, store model.Is
 							TLSMode: instance.Endpoint.TLSMode,
 						})
 					}
+					_ = c.XdsUpdater.EDSUpdate(c.Cluster(), string(instance.Service.Hostname), curr.Namespace, endpoints)
```

Comment:
Is this going to trigger an update multiple times for the same hostname if we have multiple entries for the same hostname? If so I think we will only keep the last one since endpoints will be changed each iteration

Reply:
I think we can optimize this a bit, by grouping endpoints by host. Let me make that change.,

Reply:
it's not just optimization, the second loop will undo the changes from the previous loop I think

Reply:
see if this change makes sense

Reply:
Also, updated the test case to verify that correct hosts are passed.